### ISSUE TITLE
How to change fontFamily of tabs?

### ISSUE THREAD

**Issue by ashokkumar88** [madhu314/react-native-tabbed-view-pager-android#11]
Hi,
How can I change the fontFamily of the tabs for Android?  I tried to change from `TabbedViewPager.java` but it was not working. Can u please give any idea which file should i modify to change the fontFamily.

**Comment by chiragmittal19** [comment]
@ashokkumar88 I want to do the same. Did you find any solution?

**Comment by ashokkumar88** [comment]
Yes, I found solution. I modified the Java code and added two props in js side `fontName' and `textUpperCase`. Now I am able to change the fontfamily and also able to disable uppercase.
```
fontName={tabFontName}
textUpperCase={false}
```
I will make a PR to merge it in master.

**Comment by ashokkumar88** [comment]
I have a PR here to use custom font and lower case tabnames.
https://github.com/madhu314/react-native-tabbed-view-pager-android/pull/13

**Comment by madhu314** [comment]
hey @ashokkumar88 although your PR works, its not the right way to approach this.
The correct way would be to add Spanned strings

Listen to tab selection listener, on select and unselect callbacks, you need to reset the text of the tab with Spannable string

here is code snippet from our project written in Kotlin

```
fun Tab.toSelect(context: Context) {

  val t = buildSpannedString {
    inSpans(
      CabinTypefaceSpan(context, Typeface.BOLD),
      LetterSpacingSpan(0.03f),
      ForegroundColorSpan(
        Color.parseColor("#745df1")
      )
    ) {
      append(text)
    }
  }

  text = t
}

fun Tab.toUnselect(context: Context) {

  val t = buildSpannedString {
    inSpans(
      CabinTypefaceSpan(context, Typeface.NORMAL),
      LetterSpacingSpan(0.03f),
      ForegroundColorSpan(
        Color.parseColor("#aeb4c1")
      )
    ) {
      append(text)
    }
  }

  text = t
}

fun TabLayout.selectedTab(): Tab? {
  return getTabAt(selectedTabPosition)
}
```

```
tabLayout.addOnTabSelectedListener(object : OnTabSelectedListener {

      override fun onTabReselected(tab: Tab?) {}

      override fun onTabSelected(tab: Tab?) {
        tab?.toSelect(activity())
      }

      override fun onTabUnselected(tab: Tab?) {
        tab?.toUnselect(activity())
      }
    })
```

```CabinTypefaceSpan``` is a custom typeface span.

this is not a small change and unfortunately I am not getting time to support this. I appreciate your PR

if this solution works for you, you can add the dependency from your own fork in your project.

**Comment by ashokkumar88** [comment]
But I think in the above approach the function `onTabSelected` and `onTabUnselected` will be fired multiple times whenever the tab will change. Will not it create lagging in the UI? On the other hand `onLayout` will be called only when there is layout change. Anyway currently it is working for me.

**Comment by chiragmittal19** [comment]
I have added two review comments in the pull request. Please check them out.

**Comment by chiragmittal19** [comment]
I think this issue can be closed now as it has been resolved [here](https://github.com/madhu314/react-native-tabbed-view-pager-android/issues/11#issuecomment-396486628).
